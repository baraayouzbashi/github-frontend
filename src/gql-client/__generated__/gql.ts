/* eslint-disable */
import * as types from './graphql';
import { TypedDocumentNode as DocumentNode } from '@graphql-typed-document-node/core';

/**
 * Map of all GraphQL operations in the project.
 *
 * This map has several performance disadvantages:
 * 1. It is not tree-shakeable, so it will include all operations in the project.
 * 2. It is not minifiable, so the string of a GraphQL query will be multiple times inside the bundle.
 * 3. It does not support dead code elimination, so it will add unused operations.
 *
 * Therefore it is highly recommended to use the babel or swc plugin for production.
 * Learn more about it here: https://the-guild.dev/graphql/codegen/plugins/presets/preset-client#reducing-bundle-size
 */
const documents = {
    "\n  query GetIssueWithCommentsByID($issueId: ID!, $numComments: Int!) {\n    node(id: $issueId) {\n      ... on Issue {\n        title\n        body\n        url\n        createdAt\n        state\n        author {\n          login\n        }\n        comments(first: $numComments) {\n          nodes {\n            body\n            createdAt\n            author {\n              login\n            }\n          }\n          pageInfo {\n            hasNextPage\n            endCursor\n          }\n        }\n      }\n    }\n  }\n": types.GetIssueWithCommentsByIdDocument,
    "\nquery GetRepositoryIssues($owner: String!, $name: String!, $states: [IssueState!], $IssuePagination: Int!) {\n  repository(owner: $owner, name: $name) {\n    issues(\n      states: $states\n      first: $IssuePagination\n      orderBy: {direction: DESC, field: CREATED_AT}\n    ) {\n      totalCount\n      nodes {\n        state\n        title\n        labels(first: 3) {\n          nodes {\n            color\n            name\n          }\n        }\n        comments {\n          totalCount\n        }\n        id\n      }\n    }\n  }\n}\n": types.GetRepositoryIssuesDocument,
    "\n  query SearchIssuesByTitleOrBody($query: String!, $numResults: Int!) {\n  search(query: $query, type: ISSUE, first: $numResults) {\n    edges {\n      node {\n        ... on Issue {\n          state\n          title\n          labels(first: 3) {\n            nodes {\n              color\n              name\n            }\n          }\n          comments {\n            totalCount\n          }\n          id\n          author {\n            login\n          }\n        }\n      }\n    }\n  }\n}\n": types.SearchIssuesByTitleOrBodyDocument,
};

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 *
 *
 * @example
 * ```ts
 * const query = gql(`query GetUser($id: ID!) { user(id: $id) { name } }`);
 * ```
 *
 * The query argument is unknown!
 * Please regenerate the types.
 */
export function gql(source: string): unknown;

/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(source: "\n  query GetIssueWithCommentsByID($issueId: ID!, $numComments: Int!) {\n    node(id: $issueId) {\n      ... on Issue {\n        title\n        body\n        url\n        createdAt\n        state\n        author {\n          login\n        }\n        comments(first: $numComments) {\n          nodes {\n            body\n            createdAt\n            author {\n              login\n            }\n          }\n          pageInfo {\n            hasNextPage\n            endCursor\n          }\n        }\n      }\n    }\n  }\n"): (typeof documents)["\n  query GetIssueWithCommentsByID($issueId: ID!, $numComments: Int!) {\n    node(id: $issueId) {\n      ... on Issue {\n        title\n        body\n        url\n        createdAt\n        state\n        author {\n          login\n        }\n        comments(first: $numComments) {\n          nodes {\n            body\n            createdAt\n            author {\n              login\n            }\n          }\n          pageInfo {\n            hasNextPage\n            endCursor\n          }\n        }\n      }\n    }\n  }\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(source: "\nquery GetRepositoryIssues($owner: String!, $name: String!, $states: [IssueState!], $IssuePagination: Int!) {\n  repository(owner: $owner, name: $name) {\n    issues(\n      states: $states\n      first: $IssuePagination\n      orderBy: {direction: DESC, field: CREATED_AT}\n    ) {\n      totalCount\n      nodes {\n        state\n        title\n        labels(first: 3) {\n          nodes {\n            color\n            name\n          }\n        }\n        comments {\n          totalCount\n        }\n        id\n      }\n    }\n  }\n}\n"): (typeof documents)["\nquery GetRepositoryIssues($owner: String!, $name: String!, $states: [IssueState!], $IssuePagination: Int!) {\n  repository(owner: $owner, name: $name) {\n    issues(\n      states: $states\n      first: $IssuePagination\n      orderBy: {direction: DESC, field: CREATED_AT}\n    ) {\n      totalCount\n      nodes {\n        state\n        title\n        labels(first: 3) {\n          nodes {\n            color\n            name\n          }\n        }\n        comments {\n          totalCount\n        }\n        id\n      }\n    }\n  }\n}\n"];
/**
 * The gql function is used to parse GraphQL queries into a document that can be used by GraphQL clients.
 */
export function gql(source: "\n  query SearchIssuesByTitleOrBody($query: String!, $numResults: Int!) {\n  search(query: $query, type: ISSUE, first: $numResults) {\n    edges {\n      node {\n        ... on Issue {\n          state\n          title\n          labels(first: 3) {\n            nodes {\n              color\n              name\n            }\n          }\n          comments {\n            totalCount\n          }\n          id\n          author {\n            login\n          }\n        }\n      }\n    }\n  }\n}\n"): (typeof documents)["\n  query SearchIssuesByTitleOrBody($query: String!, $numResults: Int!) {\n  search(query: $query, type: ISSUE, first: $numResults) {\n    edges {\n      node {\n        ... on Issue {\n          state\n          title\n          labels(first: 3) {\n            nodes {\n              color\n              name\n            }\n          }\n          comments {\n            totalCount\n          }\n          id\n          author {\n            login\n          }\n        }\n      }\n    }\n  }\n}\n"];

export function gql(source: string) {
  return (documents as any)[source] ?? {};
}

export type DocumentType<TDocumentNode extends DocumentNode<any, any>> = TDocumentNode extends DocumentNode<  infer TType,  any>  ? TType  : never;