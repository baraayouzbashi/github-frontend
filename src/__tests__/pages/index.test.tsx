import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import Home, { getServerSideProps } from "../../pages";
import { GetRepositoryIssuesQuery } from "@/gql-client/__generated__/graphql";
import { MockedValue } from "../../../jest.setup";

const data = {
  issues: {
    nodes: [],
    totalCount: 0,
  },
} as GetRepositoryIssuesQuery;

describe("Home", () => {
  it("renders a heading", () => {
    render(<Home data={data} />);
    const heading = screen.queryByRole("heading", { level: 1 });
    expect(heading).toBeInTheDocument();
  });
  it("renders Loading screen", () => {
    render(<Home data={[]} />);
    const loading = screen.getByText(/No results/);
    expect(loading).toBeTruthy();
  });
  it("renders homepage unchanged", () => {
    const { container } = render(<Home data={data} />);
    expect(container).toMatchSnapshot();
  });
});
describe("getServerSideProps", () => {
  it("returns data inside props", async () => {
    expect.assertions(1);
    const res = await getServerSideProps({});
    expect(res).toEqual({
      props: { data: MockedValue.repository.issues.nodes },
    });
  });
});
