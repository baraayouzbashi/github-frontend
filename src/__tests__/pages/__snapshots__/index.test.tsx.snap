// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders homepage unchanged 1`] = `
.c0 {
  max-width: 1280px;
  margin: 0 auto;
  padding: 2rem;
  font-family: -apple-system,BlinkMacSystemFont,"Segoe UI",Roboto,Helvetica,Arial,sans-serif;
}

.c1 {
  font-weight: bold;
  font-size: 3rem;
  color: var(--color-secondary);
  margin-bottom: 1rem;
}

.c3 {
  flex-grow: 1;
  padding: 0.75rem;
  font-size: 1rem;
  border: 1px solid #d1d9e0;
  border-radius: 6px;
  background: #f6f8fa;
  box-shadow: inset 0 1px 0 0 #1f23280a;
  transition: box-shadow 0.3s ease-in-out;
}

.c3:focus {
  outline: none;
  box-shadow: inset 0 0 0 1px #0969da;
}

.c2 {
  display: flex;
  flex-direction: column;
  gap: 10px;
  margin-bottom: 2rem;
}

.c2 >* {
  flex-shrink: 0;
}

.c4 {
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 5px;
}

.c7 {
  margin-top: 2rem;
  text-align: center;
}

.c5 {
  width: 42px;
  height: 25px;
  background-color: #e4e4e7;
  border-radius: 9999px;
  position: relative;
  border: 0;
}

.c5[data-state="checked"] {
  background-color: #1f883d;
}

.c6 {
  display: block;
  width: 21px;
  height: 21px;
  background-color: white;
  border-radius: 9999px;
  transition: transform 100ms;
  transform: translateX(2px);
  will-change: transform;
}

.c6[data-state="checked"] {
  transform: translateX(19px);
}

@media (min-width:786px) {
  .c2 {
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: center;
    align-items: center;
    margin-bottom: 0;
  }
}

<div>
  <div
    class="c0"
  >
    <h1
      class="c1"
    >
      facebook / react
    </h1>
    <div
      class="c2"
    >
      <input
        aria-label="Search issues"
        class="c3"
        placeholder="Search issues..."
        type="text"
        value=""
      />
      <label
        class="c4"
      >
        Show open issues
        <button
          aria-checked="false"
          class="c5"
          data-state="unchecked"
          role="switch"
          type="button"
          value="on"
        >
          <span
            class="c6"
            data-state="unchecked"
          />
        </button>
      </label>
    </div>
    <p
      class="c7"
    >
      No results ..
    </p>
  </div>
</div>
`;
